rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -25,14 +30,36 @@
  */
 public class CachedReadRequestChain extends ReadRequestChain
 {
-    private final RandomAccessFile fileToRead;
+    private FileChannel fileChannel = null;
     private int read = 0; // data read
 
+    private ByteBuffer directBuffer;
+
     private static final Log log = LogFactory.getLog(CachedReadRequestChain.class);
 
+    public CachedReadRequestChain(RandomAccessFile fileToRead, ByteBuffer buffer)
+    {
+        try {
+            FileInputStream fis = new FileInputStream(fileToRead.getFD());
+            fileChannel = fis.getChannel();
+        }
+        catch (IOException e) {
+            log.error("Unable to open file channel", e);
+            throw Throwables.propagate(e);
+        }
+        directBuffer = buffer;
+    }
+
+    @VisibleForTesting
     public CachedReadRequestChain(RandomAccessFile fileToRead)
     {
-        this.fileToRead = fileToRead;
+        this(fileToRead, ByteBuffer.allocate(1024));
+    }
+
+    @VisibleForTesting
+    public CachedReadRequestChain()
+    {
+        //Dummy constructor for testing #testConsequtiveRequest method.
```

Comment:
except in the case of testing collated read requests, we don't pass any RAF handler there.